src/hooks/bindings/useDisableUpdater.ts

```diff
@@ -44,7 +47,24 @@ function useDisableUpdater(bindingUUID?: string) {
         async (bindingIndexes: BindingDisableUpdate[]) => {
             return draftUpdater(
                 (spec) => {
+                    let missingIndex = false;
+
+                    // TODO (draft updater) - this is one of the few things that can directly
+                    //  update a binding that isn't the current binding. We should look into
+                    //  getting some of this validation shared with `ui/src/utils/workflow-utils.ts > getBindingIndex`
                     bindingIndexes.forEach(({ bindingIndex, val }) => {
+                        if (
+                            missingIndex ||
+                            // Make sure there is a binding there at least
+                            !spec.bindings[bindingIndex] ||
+                            // Try to be safe and make sure we have the right name
+                            getCollectionName(spec.bindings[bindingIndex]) ===
+                                collectionName
+                        ) {
+                            missingIndex = true;
+                            return;
+                        }
```

Comment:
Need to make sure what we are trying to disable is there. This is mainly just for `materialization` `edit` flow where a user enters the flow, adds some bindings, and then toggles the disable/enable without generating a catalog spec.